rfqmath/fixed_point_test.go

```diff
@@ -159,6 +159,144 @@ func testFromUint64[N Int[N]](t *rapid.T) {
+			withinBounds: true,
+		},
+		{
+			// Case where firstFp is 15% less than secondFp,
+			// tolerance allows 17.65% (176470 PPM). Diff within
+			// bounds.
+			firstFp:      FixedPointFromUint64[N](85_000, 3),
+			secondFp:     FixedPointFromUint64[N](100_000, 2),
+			tolerancePpm: 176470, // 17.65% tolerance in PPM
+			withinBounds: true,
+		},
+		{
+			// Case where secondFp is 15% less than firstFp,
+			// tolerance allows 10% (100000 PPM). Diff outside
+			// bounds.
+			firstFp:      FixedPointFromUint64[N](85_000, 3),
+			secondFp:     FixedPointFromUint64[N](100_000, 2),
+			tolerancePpm: 100000, // 10% tolerance in PPM
+			withinBounds: false,
+		},
+		{
+			// Case where firstFp and secondFp are equal,
+			// tolerance is 0 PPM. Diff within bounds.
+			firstFp:      FixedPointFromUint64[N](100_000, 2),
+			secondFp:     FixedPointFromUint64[N](100_000, 4),
+			tolerancePpm: 0, // 0% tolerance in PPM
+			withinBounds: true,
+		},
+		{
+			// Case where secondFp is 1% more than firstFp,
+			// tolerance allows 0.99% (9900 PPM). Diff outside
+			// bounds.
+			firstFp:      FixedPointFromUint64[N](100_000, 2),
+			secondFp:     FixedPointFromUint64[N](101_000, 3),
+			tolerancePpm: 9900, // 0.99% tolerance in PPM
+			withinBounds: false,
+		},
+		{
+			// Case where secondFp is 5% less than firstFp,
+			// tolerance allows 5% (50000 PPM). Diff within bounds.
+			firstFp:      FixedPointFromUint64[N](100_000, 1),
+			secondFp:     FixedPointFromUint64[N](95000, 2),
+			tolerancePpm: 50000, // 5% tolerance in PPM
+			withinBounds: true,
+		},
+		{
+			// Case where secondFp is 10% less than firstFp,
+			// tolerance allows 9% (90000 PPM). Diff outside bounds.
+			firstFp:      FixedPointFromUint64[N](100_000, 3),
+			secondFp:     FixedPointFromUint64[N](90_000, 1),
+			tolerancePpm: 90000, // 9% tolerance in PPM
+			withinBounds: false,
+		},
+		{
+			// Case where secondFp is 9% less than firstFp,
+			// tolerance allows 10% (100000 PPM). Diff within
+			// bounds.
+			firstFp:      FixedPointFromUint64[N](100_000, 4),
+			secondFp:     FixedPointFromUint64[N](91_000, 1),
+			tolerancePpm: 100000, // 10% tolerance in PPM
+			withinBounds: true,
+		},
+		{
+			// Case where both prices are zero, should be within
+			// bounds.
+			firstFp:      FixedPointFromUint64[N](0, 0),
+			secondFp:     FixedPointFromUint64[N](0, 0),
+			tolerancePpm: 100000, // any tolerance in PPM
+			withinBounds: true,
+		},
+		{
+			// Case where firstFp is zero and secondFp is
+			// non-zero, should not be within bounds.
+			firstFp:      FixedPointFromUint64[N](0, 0),
+			secondFp:     FixedPointFromUint64[N](100_000, 4),
+			tolerancePpm: 100000, // any tolerance in PPM
+			withinBounds: false,
+		},
+		{
+			// Case where secondFp is zero and firstFp is
+			// non-zero, should not be within bounds.
+			firstFp:      FixedPointFromUint64[N](100_000, 4),
+			secondFp:     FixedPointFromUint64[N](0, 0),
+			tolerancePpm: 100000, // any tolerance in PPM
```

Comment:
How would we pick a `tolerancePpm` value in practice?

Reply:
is this value configured by the user, to express a variable exposure to volatility?

Reply:
We currently use:
```
	AcceptPriceDeviationPpm uint64 `long:"acceptpricedeviationppm" description:"The default price deviation in parts per million that is accepted by the RFQ negotiator"`
```

as the tolerance. We will use that same config field with this new functionality.